awscmd/ecs.go

```diff
@@ -79,8 +84,90 @@ func EcsDeploy(ctx context.Context, input *InputEcsDeploy) (*OuputEcsDeploy, err
 	if err != nil {
 		return nil, fmt.Errorf("Failed to update service with new task revision: %w", err)
 	}
+	var primaryDeployment *ecs.Deployment = nil
+	for _, deployment := range updateOut.Service.Deployments {
+		if *deployment.Status == "PRIMARY" {
+			primaryDeployment = deployment
+			break
+		}
+	}
+
+	if primaryDeployment == nil {
+		return &OuputEcsDeploy{
+			Service: *updateOut.Service.ServiceName,
+		}, fmt.Errorf("Service %s deployed but couldn't fetch primary deployment status", *updateOut.Service.ServiceName)
+	}
 
 	return &OuputEcsDeploy{
-		Service: *updateOut.Service.ServiceName,
+		Service:             *updateOut.Service.ServiceName,
+		PrimaryDeploymentID: *primaryDeployment.Id,
 	}, nil
 }
+
+type InputEcsDeployWait struct {
+	Region       string
+	Cluster      string
+	Service      string
+	DeploymentID string
+}
+
+type OuputEcsDeployWait struct {
+}
+
+func EcsDeployWait(ctx context.Context, input *InputEcsDeployWait, w io.Writer) (*OuputEcsDeployWait, error) {
+	sess, err := NewSession(input.Region)
+	if err != nil {
+		return nil, err
+	}
+	svc := ecs.New(sess)
+
+	attempt := 1
+
+	for {
+		serviceOut, err := svc.DescribeServicesWithContext(ctx, &ecs.DescribeServicesInput{
+			Cluster:  aws.String(input.Cluster),
+			Services: []*string{aws.String(input.Service)},
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Failed to fetch service: %w", err)
+		}
+		if len(serviceOut.Services) == 0 {
+			return nil, fmt.Errorf("No service definition found")
+		}
+		if len(serviceOut.Services) != 1 {
+			// this should not happen because we defined only 1 service in input but stays for sanity check
+			return nil, fmt.Errorf("Ambigious match, found more than 1 service")
+		}
+
+		var primaryDeployment *ecs.Deployment = nil
```

Comment:
I'm really nitpicking on the var's name here, but can this, at this point, be a non-primary deployment? I.e., let's say a deployment finished since the last attempt, and another one started - then, if I understand correctly, this will be an active deployment rather than a primary.
If my understanding is correct, then maybe a better name would be something like "awaitedDeployment" or "monitoredDeployment"?
Again, I may be nitpicking here, but I worry that the name `primaryDeployment` may push someone in the future to change it to look for deployment with the status "PRIMARY"

Reply:
According to AWS Primary (as a type) is the active one that is about to release the app, and Active (as a type) one is the one that was deployed earlier and it's currently running (by running I mean app is running) so technically primary is fine here but I like awaited/monitored better so I will change it.